src/west/app/config.py

```diff
@@ -194,8 +206,14 @@ def read(self, args):
     def write(self, args):
         self.check_config(args.name)
         what = args.configfile or LOCAL
+        if args.append:
+            value = self.config.get(args.name,
+                                    configfile=args.configfile or ALL) or ''
```

Comment:
Does the `ALL` make sense? If the value exists at global or system level, you'd create a new local copy with the appended data.

Reply:
I thought the same thing, but my reasoning was along these lines: if the user reads the value without any additional parameters, it'll get the global or system value if there's no local one. Then, if the user intends to append, perhaps they expect to append to whatever the currently active value is when reading? I agree it's not obvious.

Reply:
Maybe appending should require an explicit target then? This hidden logic would confuse users too much IMO.

Reply:
Requiring an explicit target beats the point of this option, IMO, which is to simplify the usage of the write operation. Instead I'd rather follow the logic of write() for append() as well.

Reply:
@pdgendt I changed it now so it defaults to `LOCAL`, just like `write`.